analyzers/sdk_test.go

```diff
@@ -0,0 +1,202 @@
+	var issues []types.Issue
+
+	// trim newline from buffer output
+	lines := strings.Split(buf.String(), "\n")
+
+	for _, line := range lines {
+		// trim spaces
+		line = strings.TrimSpace(line)
+		if line == "" {
+			break
+		}
+
+		// compile regular expression for parsing unix format
+
+		// group descriptions:
+		// 0: complete string
+		// 1: path
+		// 2: line number
+		// 3: column number
+		// 4: message
+		exp, err := regexp.Compile("(.+):(.):(.): (.+)")
+		if err != nil {
+			return types.AnalysisReport{}, err
+		}
+
+		// get groups
+		groups := exp.FindAllStringSubmatch(strings.TrimSuffix(line, "\n"), -1)
+		if len(groups) == 0 {
+			return types.AnalysisReport{}, errors.New("failed to parse output string")
+		}
+
+		// convert line and column numbers to int
+		line, err := strconv.Atoi(groups[0][2])
+		if err != nil {
+			return types.AnalysisReport{}, err
+		}
+
+		col, err := strconv.Atoi(groups[0][3])
+		if err != nil {
+			return types.AnalysisReport{}, err
+		}
+
+		// compile regular expression for parsing staticcheck message
+
+		// group descriptions:
+		// 0: complete string
+		// 1: partial message string
+		// 2: issue code
+		// 3: parentheses
+		messageExp, err := regexp.Compile("(.+)[(](.+)(.+)")
+		if err != nil {
+			return types.AnalysisReport{}, err
+		}
+		messageGroups := messageExp.FindAllStringSubmatch(groups[0][4], -1)
+		if len(messageGroups) == 0 {
+			return types.AnalysisReport{}, errors.New("failed to parse message")
+		}
+
+		// populate issue
+		issue := types.Issue{
+			IssueCode: messageGroups[0][2],
+			IssueText: groups[0][4],
+			Location: types.Location{
+				Path: groups[0][1],
+				Position: types.Position{
+					Begin: types.Coordinate{
+						Line:   line,
+						Column: col,
+					},
+				},
+			},
+		}
+
+		issues = append(issues, issue)
+	}
+
+	// populate report
+	report := types.AnalysisReport{
+		Issues: issues,
+	}
+
+	// return report
+	return report, nil
+}
+
+func TestAnalyzer(t *testing.T) {
+	t.Run("wet run", func(t *testing.T) {
+		a := CLIAnalyzer{
+			Name:      "staticcheck",
+			Command:   "staticcheck",
+			Args:      []string{"-f", "text", "./testdata/triggers/staticcheck/..."},
+			Processor: &StaticCheckProcessor{},
+		}
+
+		err := a.Run()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		processedReport, err := a.Processor.Process(a.Stdout())
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// save report
+		err = utils.SaveReport(processedReport, "testdata/triggers/staticcheck/issues.json", "json")
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// read the generated report
+		reportContent, err := os.ReadFile("testdata/triggers/staticcheck/issues.json")
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		var report types.AnalysisReport
+		err = json.Unmarshal(reportContent, &report)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// do a verification check for the generated report
+		err = utils.Verify(report, "testdata/triggers/staticcheck/staticcheck.go")
```

Comment:
Can we move this from `utils` to `analysistest` package? Like https://github.com/golang/tools/tree/master/go/analysis/analysistest.